ui/lib/kubernetes/addon/components/page/roles.hbs

```diff
@@ -0,0 +1,76 @@
+<TabPageHeader @model={{@model.backend}} @filterRoles={{true}} @rolesFilterValue={{@filterValue}}>
+  <ToolbarLink @route="roles.create" @type="add" data-test-toolbar-roles-action>
+    Create role
+  </ToolbarLink>
+</TabPageHeader>
+
+{{#if (not @model.config)}}
+  <ConfigCta />
+{{else if (not @model.roles)}}
+  {{#if @filterValue}}
+    <EmptyState @title="There are no roles matching &quot;{{@filterValue}}&quot;" />
+  {{else}}
+    <EmptyState
+      data-test-config-cta
+      @title="No roles yet"
+      @message="When created, roles will be listed here. Create a role to start generating service account tokens."
+    >
+      <LinkTo class="has-top-margin-xs" @route="roles.create">
+        Create role
+      </LinkTo>
+    </EmptyState>
+  {{/if}}
+{{else}}
+  <div class="has-bottom-margin-s">
+    {{#each @model.roles as |role|}}
+      <ListItem @linkPrefix={{this.mountPoint}} @linkParams={{array "roles.role.details" role.name}} as |Item|>
+        <Item.content>
+          <Icon @name="user" />
+          {{role.name}}
+        </Item.content>
+        <Item.menu as |Menu|>
+          {{#if role.rolesPath.isLoading}}
+            <li class="action">
+              <button disabled type="button" class="link button is-loading is-transparent">
+                loading
+              </button>
+            </li>
+          {{else}}
+            <li class="action">
+              <LinkTo
+                class="has-text-black has-text-weight-semibold"
+                data-test-details
+                @route="roles.role.details"
+                @model={{role}}
+                @disabled={{not role.canRead}}
```

Comment:
I remember @chelshaw had recommended I refactor and use `{{eq role.canRead false}}` for link capabilities in the past. This way the default behavior is enabling the link so we avoid disabling an action a user may have access to if for some reason this capability is `undefined`. 

since this is wrapped in the `isLoading` conditional `{{not role.canRead}}` is probably sufficient. I can't think of when else it might be `undefined` except for older models 🤔 - anyway, just adding for context!

Reply:
Oh great thanks for the context! I didn't consider the third state and it makes sense to ignore if `undefined` or `null` and then the API would handle the permissions error if need be. I'm going to update to explicitly check for `false` to be safe.